### ISSUE TITLE
Finetune_your_model 

### ISSUE THREAD

**Issue by xuzero** [yjxiong/anet2016-cuhk#6]
Hi ,Thanks for  sharing your resnet model!  I  want to finetune the resnet 200 model on my own rgb dataset which consists of 20 classes. i change the deploy file to train-val.prototxt, add 4 layers: Train VideoData layer，Test VideoData layer, prob layer , perbt layer.
This twoVideoData layers crop_size is 224. I think they are ok.
Details are:
layer {
  name: "lkaction"
  type: "InnerProduct"
  bottom: "caffe.Flatten_673"
  top: "lkaction"
  inner_product_param {
    num_output: 20
    axis: -1    
   weight_filler {
      		  type: "xavier"
    		}
    		bias_filler {
      		  type: "constant"
      		  value: 0
    	        }    	        
  }
}
layer {
	bottom: "lkaction"
	bottom: "label"
	top: "prob"
	name: "prob"
	type: "SoftmaxWithLoss"
	include {
	  phase: TRAIN
	}
}

layer {
       name: "probt"
       type: "Softmax"
       bottom: "lkaction"
       top: "probt"
       include {
	   phase: TEST
	}
}

The log says that
I1111 12:46:10.121731 38914 caffe.cpp:86] Finetuning from /home/lk/anet2016-cuhk/models/resnet200_anet_2016.caffemodel
F1111 12:46:10.761404 38915 blob.cpp:455] Check failed: ShapeEquals(proto) shape mismatch (reshape not set)

Could you please tell me how to debug it?

**Comment by xuzero** [comment]
The details of the two VideoData layers  are:
layer {
  name: "resnet_200"
  type: "VideoData"
  top: "data"
  top: "label"
 
    video_data_param {
    source: "/home/lk/anet2016-cuhk/data/shuffle_val_leyframe_list.txt"
    batch_size: 8
    new_width: 340
    new_height: 256
    new_length: 1
    num_segments: 3
    shuffle: true
    modality: RGB
  }
  transform_param{
    crop_size: 224
    mirror: true
    fix_crop: true
    multi_scale: true
    more_fix_crop: true
    max_distort: 1
    scale_ratios: [1,.875,.75,.66]
     is_flow: false
    mean_value: [104, 117, 123, 104, 117, 123, 104, 117, 123]
  }
  include: { phase: TRAIN }

layer {
  name: "resnet_200"
  type: "VideoData"
  top: "data"
  top: "label"
 
    video_data_param {
    source: "/home/lk/anet2016-cuhk/data/shuffle_val_leyframe_list.txt"
    batch_size: 1
    new_width: 340
    new_height: 256
    new_length: 1
    num_segments: 3
    shuffle: true
    modality: RGB
  }
  transform_param{
    crop_size: 224  
     is_flow: false
    mean_value: [104, 117, 123, 104, 117, 123, 104, 117, 123]
  }
  include: { phase: TEST }  
}

Thanks for your help!

**Comment by yjxiong** [comment]
If you look at the initialization log, you can find the output size of each layer, in the format of 

```
xxx - > xxx
Setting up xxx
Top shape: a b c d
```

By examining these numbers, it might be easier for you to identify the issue. 

Or if you have any problem locating the problem, you can also put the initialization here, I may help you look at it.

**Comment by xuzero** [comment]
The log file linking is http://pan.baidu.com/s/1gfeCCoF  
Thanks for your help!

**Comment by xuzero** [comment]
The solver.prototxt and train_val.prototxt links are http://pan.baidu.com/s/1jInZX8i and http://pan.baidu.com/s/1i5Hiofz

**Comment by xuzero** [comment]
Hi，I find something about the bug ,if i set the num_segments to 1 ,the model can be finetuned but the loss is almost not change. If i set the num_segments to 3 ,the bug will appear again.
Could you please tell me why? Thank you！

**Comment by yjxiong** [comment]
You need to add this line to train the model with TSN. 

https://github.com/yjxiong/temporal-segment-networks/blob/master/models/ucf101/tsn_bn_inception_rgb_train_val.prototxt#L49

Also, there are other parts that has to be changed. Please consult the linked file.

**Comment by dmvictor** [comment]
Hi, @xuzero 
I want to finetune the Resnet 200 model on my RGB dataset which consists of 22 classes. I met the same error as yours. Could you mind sharing how you solved the problem? [This](https://gist.github.com/dmvictor/66ab6e0ec6ff674600a791a070e567dc) is my prototxt for training. [Reshape layer](https://gist.github.com/dmvictor/66ab6e0ec6ff674600a791a070e567dc#file-tsn_resnet200_rgb_train_val-prototxt-L49) is already added in the prototxt. Thank you!
